core/src/validator.rs

```diff
@@ -659,35 +680,36 @@ fn backup_and_clear_blockstore(blockstore: &Arc<Blockstore>, ledger_path: &Path,
                 );
                 last_print = Instant::now();
             }
+            last_slot = Some(slot);
         }
 
-        info!("Purging slots {} to {}", start_slot, end_slot);
-        blockstore.purge_slots_with_delay(start_slot, end_slot, None, PurgeType::Exact);
-        info!("Purging done, compacting db..");
-        if let Err(e) = blockstore.compact_storage(start_slot, end_slot) {
-            warn!(
-                "Error from compacting storage from {} to {}: {:?}",
-                start_slot, end_slot, e
-            );
+        if let Some(end_slot) = last_slot {
```

Comment:
This seems guaranteed to be `Some` at this point, can we unwrap()?

Reply:
yes, changed.